src/dmd/dcast.d

```diff
@@ -3827,3 +3835,58 @@ IntRange getIntRange(Expression e)
     e.accept(v);
     return v.range;
 }
+
+Type checkArrayOpType(Expression e1, Expression e2, TOK op, Scope *sc)
```

Comment:
public functions need to be documented, however, it seems that this function could be private. Either way, please document it using the [ddoc style](https://dlang.org/spec/ddoc.html).